collect_app/src/test/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManagerTest.java

```diff
@@ -56,38 +66,51 @@
 
     private void stubSavedAccount(String accountName) {
         when(mockPreferences.get(PreferenceKeys.KEY_SELECTED_GOOGLE_ACCOUNT)).thenReturn(accountName);
+        stubAccount(accountName);
     }
 
     private void stubCredential() {
-        doAnswer(new Answer() {
-            @Override
-            public Object answer(InvocationOnMock invocation) throws Throwable {
-                currentAccount = invocation.getArgument(0);
-                return null;
-            }
+        doAnswer(invocation -> {
+            currentAccount = invocation.getArgument(0);
+            return null;
         }).when(mockedCredential).setSelectedAccountName(anyString());
     }
 
+    private void stubAccount(String name) {
+        Account account = mock(Account.class);
+        Whitebox.setInternalState(account, "name", name);
+        doReturn(new Account[]{account}).when(mockedCredential).getAllAccounts();
+    }
+
+    private void removeAccounts() {
+        doReturn(null).when(mockedCredential).getAllAccounts();
+    }
+
+    private void mockPermissionUtils() {
+        mockStatic(PermissionUtils.class, invocation -> {
+            Whitebox.invokeMethod(googleAccountsManager, "chooseAccount");
```

Comment:
This was done because some of the approaches I was trying to utilize to invoke the direct `requestGetAccountsPermission` weren't working and I just decided that within this context the only function of the `PermissionsUtil` that we care about is this one so it's okay if an `Answer` just invokes the method that follows the permission being granted.

Reply:
Directly invoking the intended method smells like coupled code to me. What if `GoogleAccountManager` implements `PermissionListener` and then instead of invoking the method using reflection, we do,
```
PermissionListener permissionListener = googleAccountsManager();
permissionListener.onGranted()
```

Reply:
Thats true! The reason why it had to be done like this is `PermissionUtils` contain static methods that encapsulate the permissions management provided by the `Dexter` library. The listeners aren't directly implemented on calling component so I am not able to get a referene to it as you described so that's the reason this approach was taken.

Reply:
The design decisions around the permissions code made it easy to replicate the changes across the codebase in a simpler manner without having the calling being tangled to the permissions logic. That was the idea behind it.